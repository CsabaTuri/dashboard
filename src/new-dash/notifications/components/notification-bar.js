import React from "react"
import { connect } from "react-redux"
import { MessageBar, MessageBarType } from "office-ui-fabric-react"

import "./notification-bar.css"
import { NotificationType } from "../"

const messageBarTypeFor = (notificationType) => {
  switch (notificationType) {
    case NotificationType.ERROR:   return MessageBarType.error
    case NotificationType.SUCCESS: return MessageBarType.success
    default:                       return MessageBarType.info
  }
}

const NotificationBar = (props) => {
  return <div className="notification-bar">
      <ul>{
      props.notifications.map((notification, index) => (
        <li key={index} className="ms-u-fadeIn100">
          <MessageBar
            messageBarType={messageBarTypeFor(notification.type)}>
            {notification.message}
          </MessageBar>
        </li>
      ))
    }</ul>
  </div>
}

export default connect(
  state => ({
    notifications: state.get("notifications")
  })
)(NotificationBar)

