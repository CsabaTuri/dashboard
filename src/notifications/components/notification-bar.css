.notification-bar ul {
    list-style: none;
    margin: 0;
    padding: 0;
}

.notification-bar li {
    padding: 2px 0;
}
