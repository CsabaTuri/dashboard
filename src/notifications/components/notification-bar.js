import React from "react"
import { connect } from "react-redux"
import { MessageBar, MessageBarType } from "office-ui-fabric-react"

import "./notification-bar.css"
import { NotificationType } from "../"

const messageBarTypeFor = (notificationType) => {
  switch (notificationType) {
    case NotificationType.ERROR:   return MessageBarType.error
    case NotificationType.SUCCESS: return MessageBarType.success
    default:                       return MessageBarType.info
  }
}

export const NotificationBar = ({ notifications }) => {
  return <div className="notification-bar ms-Grid-col ms-u-sm12 ms-u-md6 ms-u-mdPush3">
      <ul>{
      notifications.map((notification, index) => (
        <li key={index} className="ms-u-fadeIn100">
          <MessageBar
            messageBarType={messageBarTypeFor(notification.get("type"))}>
            {notification.get("message").split("\n").map((text, key) =>
              <span key={key}>{text}<br/></span>
            )}
          </MessageBar>
        </li>
      ))
    }</ul>
  </div>
}

NotificationBar.displayName = "NotificationBar"

export default connect(
  state => ({
    notifications: state.get("notifications")
  })
)(NotificationBar)

