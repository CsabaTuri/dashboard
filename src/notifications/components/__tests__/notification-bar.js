import React from "react"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import { List, Map } from "immutable"

import { NotificationBar } from "../notification-bar"

it("renders empty", () => {
  const comp = shallow(<NotificationBar notifications={List()}/>)
  expect(shallowToJson(comp)).toMatchSnapshot()
})

it("displays notifications", () => {
  const comp = shallow(
    <NotificationBar notifications={List.of(
      Map.of("type", "error", "message", "an error"),
      Map.of("type", "success", "message", "a success"),
      Map.of("type", "info", "message", "an info")
    )}/>
  )

  expect(shallowToJson(comp)).toMatchSnapshot()
})
