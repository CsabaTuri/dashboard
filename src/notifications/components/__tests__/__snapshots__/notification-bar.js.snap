exports[`test displays notifications 1`] = `
<div
  className="notification-bar ms-Grid-col ms-u-sm12 ms-u-md6 ms-u-mdPush3">
  <ul>
    <li
      className="ms-u-fadeIn100">
      <MessageBar
        isMultiline={true}
        messageBarType={1}
        onDismiss={null}>
        <span>
          an error
          <br />
        </span>
      </MessageBar>
    </li>
    <li
      className="ms-u-fadeIn100">
      <MessageBar
        isMultiline={true}
        messageBarType={4}
        onDismiss={null}>
        <span>
          a success
          <br />
        </span>
      </MessageBar>
    </li>
    <li
      className="ms-u-fadeIn100">
      <MessageBar
        isMultiline={true}
        messageBarType={0}
        onDismiss={null}>
        <span>
          an info
          <br />
        </span>
      </MessageBar>
    </li>
  </ul>
</div>
`;

exports[`test renders empty 1`] = `
<div
  className="notification-bar ms-Grid-col ms-u-sm12 ms-u-md6 ms-u-mdPush3">
  <ul />
</div>
`;
